addons/stock_account/models/account_move.py

```diff
@@ -255,6 +256,8 @@ def _eligible_for_cogs(self):
         return self.product_id.type == 'product' and self.product_id.valuation == 'real_time'
 
     def _get_gross_unit_price(self):
+        if not self.quantity:
```

Comment:
Hello, We had to recompute the price unit from the whole quantity in this [commit](https://github.com/odoo/odoo/commit/360b3aae3a66a369c6aecbf3f1fac6a05cba1288) due to rounding errors. I would instead fall back on `self.price_unit` in case of `quantity == 0` to be sure we do not block a potential flow that require setting invoice lines to 0 quantity.

Reply:
Hello @Whenrow,

Regarding your concern, if there are any potential flaws such as service charges, the "_get_gross_unit_price" method will not be called. This method is used to calculate the gross unit price of a product and we cannot calculate it with a quantity of 0.

However, please let me know if there is any scenario I need to test and act on.
@pfertyk

Reply:
@hs-oerp I don't have any specific use case in mind. I just think it's better to make sure the 0 qty flow is feasible without error, like it was before the rounding fix.

Reply:
Hello @Whenrow , Service charges often involve use cases with zero quantity. However, when the quantity is zero, the rounding fix will result in an error (_float division by zero_) at the first line where the price unit is calculated (_price_unit = self.price_subtotal / self.quantity_). Therefore, before implementing a rounding solution, it is important to confirm the quantity to avoid any such errors.

Reply:
@hs-oerp I'm confirming the rounding fix is wrong, and we can't divide by 0. I'm wondering why the fallback solution is not good to you compare to the blocking one.

Reply:
Hello @Whenrow Here is the explanation 
In our case, we have a stock journal entry and a vendor bill that need to be reconciled with each other. 

If we use the formula _(price_unit = self.quantity and self.price_subtotal / self.quantity)_, Odoo will create a bill with 0 and we will end up with two journal entries (vendor bill and stock journal entry). Odoo will not reconcile them, and they will remain as standalone entries.

However, if we use the price 0.0 logic in this case, Odoo will reconcile both entries with each other. This will not affect the numbers much, but it will ensure that the entries are reconciled.

So that I am blocking the user and forcing them to add a quantity greater than 0 and proceed with a 0.0 price.

Reply:
Hello @Whenrow 
Please let me know if there is anything else I need to do regarding this PR.

Reply:
What about `price_unit = self.quantity and self.price_subtotal / self.quantity or self.price_unit` formula?

Reply:
Hello @MiquelRForgeFlow 
Thank you for the feedback. Here is my explanation.

In this case, we have a stock journal entry and a vendor bill that need to be reconciled with each other.

If we use the formula (_price_unit = self.quantity and self.price_subtotal / self.quantity_), Odoo will create a bill with 0 and we will end up with two journal entries (vendor bill and stock journal entry). Odoo will not reconcile them, and they will remain as standalone entries.

However, if we use the price 0.0 logic in this case, Odoo will reconcile both entries with each other. This will not affect the numbers much, but it will ensure that the entries are reconciled.

So that I am blocking the user and forcing them to add a quantity greater than 0 and proceed with a 0.0 price.

Let me know if I need to change this logic.

Reply:
Thanks for the copy pasta, but I asked a different formula. I added `or self.price_unit`, so it's not the same.